### PR TITLE
Feature/monitoring intialization policy

### PR DESCRIPTION
**Pull request by kiuby88**
Policy to configure SeaClouds Datacollector.
The policy generation was added to DamGenerator too.
@rosogon , could you take a look please?

### CODE REVIEW

**Comment by rosogon** [comment]
@kiuby88 : Please, check travis fail.

File: planner/core/src/main/java/eu/seaclouds/platform/planner/core/application/ApplicationFacade.java [hunk 1]
```diff
@@ -209,4 +212,16 @@ private void updateNodeTypes(Map<String, Object> newNodeTypes) {
         setNodeTypes(usedNodeTemplates);
     }
 
+    public List<String> filterNodeTypesByPropertyAndValues(String property, List<Object> allowedValues){
```
Comment (rosogon) [inline planner/core/src/main/java/eu/seaclouds/platform/planner/core/application/ApplicationFacade.java]: Maybe filterNodeTemplatesByPropertyAndValues()?

File: planner/core/src/test/java/eu/seaclouds/platform/planner/core/application/agreements/AgreementGeneratorTest.java [hunk 2]
```diff
@@ -1,79 +1,68 @@
...
 
 import eu.seaclouds.monitor.monitoringdamgenerator.MonitoringInfo;
+import it.polimi.tower4clouds.rules.MonitoringRules;
 
 public class AgreementGeneratorTest {
 
+
+    private MockWebServer server = new MockWebServer();
+
+    @BeforeMethod(alwaysRun = true)
```
Comment (rosogon) [inline planner/core/src/test/java/eu/seaclouds/platform/planner/core/application/agreements/AgreementGeneratorTest.java]: Oh, much nicer :)
Thanks

**Comment by rosogon** [comment]
Please, modify the PR comment to describe the policy being added to the DAM. It would be much easier to follow.

**Comment by kiuby88** [comment]
@rosogon thanks for your review.  Take a look

File: deployer/src/test/java/eu/seaclouds/policy/AbstractSeaCloudsPoliciesLiveTests.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    public static final String SLA_ENDPOINT = "http://52.36.119.104:9003";
+    public static final String T4C_ENDPOINT = "http://52.48.187.2:8170";
+    public static final String INFLUXDB_ENDPOINT = "http://52.48.187.2:8086";
+    public static final String INFLUXDB_DATABASE = "tower4clouds";
+    public static final String INFLUXDB_USERNAME = "root";
+    public static final String INFLUXDB_PASSWORD = "root";
+    public static final String GRAFANA_ENDPOINT = "http://52.48.187.2:3000";
+    public static final String GRAFANA_USERNAME = "admin";
+    public static final String GRAFANA_PASSWORD = "admin";
+    public static final String SEACLOUDS_DC_ENDPOINT = "http://52.48.187.2:8176/";
```
Comment (perezp) [inline deployer/src/test/java/eu/seaclouds/policy/AbstractSeaCloudsPoliciesLiveTests.java]: Might these tests be dangerous if InfluxDB/Grafana/T4C/SLA/... components are deployed in the future in a different IP/port ?

File: deployer/pom.xml [hunk 4]
```diff
@@ -54,6 +54,10 @@
             <artifactId>brooklyn-software-webapp</artifactId>
         </dependency>
         <dependency>
+            <groupId>org.apache.brooklyn</groupId>
+            <artifactId>brooklyn-camp</artifactId>
```
Comment (perezp) [inline deployer/pom.xml]: A new dependency to brooklyn-camp. Is necessary a dependency to camp?

File: deployer/src/test/java/eu/seaclouds/policy/AbstractSeaCloudsPoliciesLiveTests.java (hunk 3 above)
Comment (kiuby88) [inline deployer/src/test/java/eu/seaclouds/policy/AbstractSeaCloudsPoliciesLiveTests.java]: I agree, probably a MockTest will be better here.

File: deployer/pom.xml (hunk 4 above)
Comment (kiuby88) [inline deployer/pom.xml]: `SeaCloudsMonitoringInitializationPolicies` requires to find the application children which are pointed by the children Id configkey [TARGET_ENTITIES](https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/300/files#diff-64d88bf2a01ddc5c605f6710156bfdbaR65). Ids can be added to the entities using two different configkeys either [TOSCA_PLAN_ID](https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/300/files#diff-64d88bf2a01ddc5c605f6710156bfdbaR61) or [CAMP_PLAN_ID](https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/300/files#diff-64d88bf2a01ddc5c605f6710156bfdbaR108) so I added CAMP dependency to use this constant but as you mentioned this includes more dependencies, and I could remove it.

File: deployer/src/test/java/eu/seaclouds/policy/AbstractSeaCloudsPoliciesLiveTests.java (hunk 3 above)
Comment (rosogon) [inline deployer/src/test/java/eu/seaclouds/policy/AbstractSeaCloudsPoliciesLiveTests.java]: I usually supply the endpoints of these live tests with env vars (if not suplied, the test does nothing). I am afraid this solution won't work in travis, but it is always interesting to have real integration tests.

**Comment by rosogon** [comment]
+1

**Pull request merged**